libraries/chain/apply_context.cpp

```diff
@@ -541,14 +620,39 @@ void apply_context::schedule_deferred_transaction( const uint128_t& sender_id, a
                subjective_block_production_exception,
                "Cannot charge RAM to other accounts during notify."
    );
-   add_ram_usage( payer, (config::billable_size_v<generated_transaction_object> + trx_size) );
+   add_ram_usage( payer, (config::billable_size_v<generated_transaction_object> + trx_size), ram_trace(get_action_id(), event_id.c_str(), "deferred_trx", operation.c_str(), "deferred_trx_add") );
 }
 
 bool apply_context::cancel_deferred_transaction( const uint128_t& sender_id, account_name sender ) {
+
+
    auto& generated_transaction_idx = db.get_mutable_index<generated_transaction_multi_index>();
    const auto* gto = db.find<generated_transaction_object,by_sender_id>(boost::make_tuple(sender, sender_id));
    if ( gto ) {
-      add_ram_usage( gto->payer, -(config::billable_size_v<generated_transaction_object> + gto->packed_trx.size()) );
+      fc::string event_id;
+      if (auto dm_logger = control.get_deep_mind_logger()) {
+         // unpack gtx->packed_trx into `dtrx` REVISE THIS!
+         auto gtx = generated_transaction(*gto);
```

Comment:
Why are you copying from `gto` rather than just using `gto` directly? All usage of `gto` for logging should be done prior to it being removed.

Reply:
Not sure why we had this part, I removed it, did not see any problems from our test suite.